gateway/src/main/java/io/zeebe/gateway/impl/broker/cluster/BrokerTopologyManagerImpl.java

```diff
@@ -15,57 +15,50 @@
  */
 package io.zeebe.gateway.impl.broker.cluster;
 
-import io.zeebe.gateway.impl.broker.request.BrokerTopologyRequest;
-import io.zeebe.gateway.impl.broker.response.BrokerResponse;
-import io.zeebe.protocol.impl.data.cluster.TopologyResponseDto;
+import static io.zeebe.gateway.impl.broker.BrokerClientImpl.LOG;
+
+import com.esotericsoftware.minlog.Log;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.atomix.cluster.ClusterMembershipEvent;
+import io.atomix.cluster.ClusterMembershipEvent.Type;
+import io.atomix.cluster.ClusterMembershipEventListener;
+import io.atomix.cluster.Member;
+import io.zeebe.raft.state.RaftState;
 import io.zeebe.transport.ClientOutput;
-import io.zeebe.transport.ClientResponse;
-import io.zeebe.transport.ClientTransport;
 import io.zeebe.transport.SocketAddress;
 import io.zeebe.util.sched.Actor;
-import io.zeebe.util.sched.clock.ActorClock;
 import io.zeebe.util.sched.future.ActorFuture;
-import io.zeebe.util.sched.future.CompletableActorFuture;
-import java.time.Duration;
-import java.util.ArrayList;
-import java.util.List;
+import java.io.IOException;
+import java.util.Properties;
+import java.util.Set;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.BiConsumer;
-import java.util.function.Consumer;
-
-public class BrokerTopologyManagerImpl extends Actor implements BrokerTopologyManager {
-  /** Interval in which the topology is refreshed even if the client is idle */
-  public static final Duration MAX_REFRESH_INTERVAL_MILLIS = Duration.ofSeconds(10);
-
-  /**
-   * Shortest possible interval in which the topology is refreshed, even if the client is constantly
-   * making new requests that require topology refresh
-   */
-  public static final Duration MIN_REFRESH_INTERVAL_MILLIS = Duration.ofMillis(300);
 
+public class BrokerTopologyManagerImpl extends Actor
+    implements BrokerTopologyManager, ClusterMembershipEventListener {
   protected final ClientOutput output;
   protected final BiConsumer<Integer, SocketAddress> registerEndpoint;
-
   protected final AtomicReference<BrokerClusterStateImpl> topology;
-  protected final List<CompletableActorFuture<BrokerClusterState>> nextTopologyFutures =
-      new ArrayList<>();
-
-  protected final BrokerTopologyRequest topologyRequest = new BrokerTopologyRequest();
-
-  protected int refreshAttempt = 0;
-  protected long lastRefreshTime = -1;
+  private final ObjectMapper objectMapper;
 
   public BrokerTopologyManagerImpl(
       final ClientOutput output, final BiConsumer<Integer, SocketAddress> registerEndpoint) {
     this.output = output;
     this.registerEndpoint = registerEndpoint;
 
-    this.topology = new AtomicReference<>(null);
+    this.objectMapper = new ObjectMapper();
+    this.topology = new AtomicReference<>(new BrokerClusterStateImpl());
```

Comment:
The initial topology reference should be null, to indicate that the topology is not ready yet. We use this distinction in places where we required information like the partition count: https://github.com/zeebe-io/zeebe/blob/develop/gateway/src/main/java/io/zeebe/gateway/impl/broker/RoundRobinDispatchStrategy.java#L35

This is also important to only publish messages after we determined the correct partition count, otherwise messages could end up on the wrong partition.